pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -25,13 +27,14 @@ import (
 const failedSetCPUManagerLabelFmt = "failed to set a cpu manager label on host %s"
 
 type HeartBeat struct {
-	clientset                 k8scli.CoreV1Interface
-	deviceManagerController   device_manager.DeviceControllerInterface
-	clusterConfig             *virtconfig.ClusterConfig
-	host                      string
-	cpuManagerPaths           []string
-	devicePluginPollIntervall time.Duration
-	devicePluginWaitTimeout   time.Duration
+	clientset                           k8scli.CoreV1Interface
+	deviceManagerController             device_manager.DeviceControllerInterface
+	clusterConfig                       *virtconfig.ClusterConfig
+	host                                string
+	cpuManagerPaths                     []string
+	devicePluginPollIntervall           time.Duration
+	devicePluginWaitTimeout             time.Duration
+	isVirtualizationSupportedOnNodeFunc func() bool
```

Comment:
If this variable is added to the heartbeat struct just to be able to change it in the unittest I think a better way to do it is like done here: https://github.com/kubevirt/kubevirt/blob/main/pkg/virtctl/vmexport/vmexport.go#L113
i.e create a func variable for it that can be changed in the UT.. since they are in the same package it can still be private

Reply:
Can you explain why do you find it better? I think this approach has the following advantages:
1) Ability to test in parallel. This isn't possible if test cases are mutating a global variable. Today most of our unit tests don't run in parallel by default, but I'm planning to add a `make test-parallel` or something similar in the future.
2) Some packages contain many structs that perform different tasks - in these packages it might be very confusing to have a bunch of function-variables. Adding the relevant function to a specific task gives clarity of the scope of which this function is expected to be used, therefore more readable imo. This package is relatively small, but it can grow in the future.

Reply:
I get your points, specially the parallel thing. I still think adding a variable to a struct just for UT is not the best :) but that's ok